psh/test-mkdir.py

```diff
@@ -0,0 +1,143 @@
+import random
+import string
+
+from collections import namedtuple
+
+File = namedtuple('File', ['name', 'owner', 'is_dir'])
+
+def ls(p, dir=''):
+    files = []
+
+    p.sendline(f'ls -la {dir}')
+    p.expect_exact('ls -la')
+    if dir:
+        p.expect_exact(f' {dir}')
+
+    p.expect_exact('\n')
+
+    while True:
+        idx = p.expect(['\(psh\)\% ', '(.*)(\r+)\n'])
+        if idx == 0:
+            break
+
+        line = p.match.group(0)
+
+        permissions, _, owner, _, _, _, _, _, name = line.split()
+        # Line is printed with ascii escaped characters - remove them
+        if name.startswith('\x1b[34m'):
+            name = name[5:]
+        if name.endswith('\x1b[0m'):
+            name = name[:-4]
+
+        f = File(name, owner, permissions[0] == 'd')
+        files.append(f)
+
+    return files
+
+def mkdir(p, dir):
+    p.sendline(f'mkdir {dir}')
+    p.expect_exact(f'mkdir {dir}')
+    p.expect_exact('\n')
+
+    idx = p.expect(['\(psh\)\% ', 'mkdir: failed to create (.*) directory'])
+
+    if idx == 1:
+        # Failed, read one more time (psh)%
+        p.expect_exact('(psh)% ')
+
+    return idx != 1
+
+def dir_present(dir, files):
+    for file in files:
+        if dir == file.name:
+            if file.owner != 'root':
+                print(f'psh ls: {dir} not owned by root')
+                return False
+            if not file.is_dir:
+                print(f'psh ls: {dir} is not dictionary')
+                return False
+
+            break
+    else:
+            print(f'psh ls: failed to found {dir}')
+            return False
+
+    return True
+
+def assert_dir_created(p, dir):
+    if not mkdir(p, dir):
+        print(f'psh mkdir: failed to create {dir}')
+        return False
+
+    path = dir.rsplit('/', 1)
+    if len(path) == 2:
+        dir = path[1]
+        path = path[0]
+    else:
+        path = '/'
+
+    files = ls(p, path)
+
+    return dir_present(dir, files)
+
+def random_dirs(p, pool, count=20):
+    if not mkdir(p, '/random_dirs'):
+        print(f'psh mkdir: failed to create /random_dirs')
+        return False
+
+    dirs = {''.join(random.choices(pool, k=random.randint(8, 16))) for _ in range(count)}
+    for dir in dirs:
+        if not mkdir(p, '/random_dirs/' + dir):
+            print(f'psh mkdir: failed to create {dir}')
+            return False
+
+    files = ls(p, '/random_dirs')
+
+    if len(files) > len(dirs) + 2:      # +2 because of . and ..
```

Comment:
hmm... this might fail if we would be running the same test 2 times on the same image (where `/random_dirs` will be on persistent storage (like ext2 partition)). Maybe we should remove the dir at the beginning of the test?

Reply:
I'd suggest removing test folders **after** we verify they were successfully created by mkdir command. We should cleanup the mess we create.

Reply:
There is no `rm` command in psh.

Reply:
@Maxez This was my initial idea, but currently we don't have "cleanup" hook (some code which would be called regardless of the test outcome) - currently if the test would fail or raise an exception we would need complicated code constructs to always cleanup. Because of that I proposed to "clean at the beginning" for this particular test :).

This is not a merge blocker as potential FS changes during testing and how to address them is a subject for broader discussion ;).